hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -0,0 +1,533 @@
+        return True
+
+    @property
+    def is_trading_required(self) -> bool:
+        return self._trading_required
+
+    def supported_order_types(self) -> List[OrderType]:
+        """
+        :return a list of OrderType supported by this connector.
+        Note that Market order type is no longer required and will not be used.
+        """
+        return [OrderType.LIMIT]
+
+    def _is_request_exception_related_to_time_synchronizer(self, request_exception: Exception):
+        return "time" in str(request_exception)
+
+    def _create_web_assistants_factory(self) -> WebAssistantsFactory:
+        return web_utils.build_api_factory(
+            throttler=self._throttler, time_synchronizer=self._time_synchronizer, auth=self._auth
+        )
+
+    def _create_order_book_data_source(self) -> OrderBookTrackerDataSource:
+        return BloxrouteOpenbookAPIOrderBookDataSource(
+            provider=self._provider_1, trading_pairs=self._trading_pairs, connector=self
+        )
+
+    def get_order_book(self, trading_pair: str) -> OrderBook:
+        blxr_ob, timestamp = self._order_manager.get_order_book(trading_pair)
+        snapshot_msg: OrderBookMessage = BloxrouteOpenbookOrderBook.snapshot_message_from_exchange(
+            msg={
+                "orderbook": blxr_ob,
+            },
+            timestamp=timestamp,
+            metadata={"trading_pair": trading_pair},
+        )
+
+        ob = BloxrouteOpenbookOrderBook()
+        ob.apply_snapshot(snapshot_msg.bids, snapshot_msg.asks, snapshot_msg.update_id)
+
+        return ob
+
+    def _create_user_stream_data_source(self) -> UserStreamTrackerDataSource:
+        return UserStreamTrackerDataSource()
+
+    def _is_user_stream_initialized(self):
+        return True
+
+    def _create_user_stream_tracker(self):
+        return None
+
+    def _create_user_stream_tracker_task(self):
+        return None
+
+    def _get_fee(
+        self,
+        base_currency: str,
+        quote_currency: str,
+        order_type: OrderType,
+        order_side: TradeType,
+        amount: Decimal,
+        price: Decimal = s_decimal_NaN,
+        is_maker: Optional[bool] = None,
+    ) -> TradeFeeBase:
+        """
+        To get trading fee, this function is simplified by using fee override configuration. Most parameters to this
+        function are ignore except order_type. Use OrderType.LIMIT_MAKER to specify you want trading fee for
+        maker order.
+        """
+
+        return DeductedFromReturnsTradeFee(percent=Decimal(0))
+
+    async def _place_order(
+        self,
+        order_id: str,
+        trading_pair: str,
+        amount: Decimal,
+        trade_type: TradeType,
+        order_type: OrderType,
+        price: Decimal,
+        **kwargs,
+    ) -> Tuple[str, float]:
+        side = api.Side.S_BID if trade_type == TradeType.BUY else api.Side.S_ASK
+        type = api.OrderType.OT_LIMIT if order_type == OrderType.LIMIT else api.OrderType.OT_MARKET
```

Comment:
avoid variables named `type` –– it's a built in python function. something liek `typ` works